test_stack.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import unicode_literals
+import pytest
+from stack import Stack
+
+@pytest.fixture()
+def empty_stack():
+    stack = Stack()
+    return stack
+
+
+@pytest.fixture()
+def full_stack():
+    stack = Stack([1, 2, 'a', 3, None, 'foobar'])
+    return stack
+
+
+def test_constructor(empty_stack, full_stack):
+    assert isinstance(empty_stack, Stack)
+    assert isinstance(full_stack, Stack)
+
+
+def test_pop(empty_stack, full_stack):
+    with pytest.raises(LookupError):
+        val = empty_stack.pop()
+
+    val = full_stack.pop()
+    assert val == 'foobar'
+    val = full_stack.pop()
+    assert val is None
+    val = full_stack.pop()
+    assert val == 3
```

Comment:
Do you see the implicit "knowledge" of the test setup in your `full_stack` fixture that this test has?  Why should your test code know anything about what values were placed into that stack?  Could you figure out a way to pass that knowledge explicitly, so that you can stop relying on implementation details of your fixture?